config.tpl.yml

```diff
@@ -145,16 +145,10 @@ APPS:
     min_instances: {{ MIN_INSTANCE_COUNT_TEMPLATE_PREVIEW }}
     max_instances: {{ MAX_INSTANCE_COUNT_TEMPLATE_PREVIEW }}
     scalers:
-      - type: ScheduledJobsScaler
-        threshold: 8
-      - type: SqsScaler
-        queues:  [create-letters-pdf-tasks]
-        # 5 gunicorn workers per * 60 seconds * pessimistic 1 second per request avg = 300 requests per minute
-        tasks_per_worker_per_minute: 300
-        allowed_queue_backlog_per_worker: 100
       - type: ScheduleScaler
         schedule:
-          scale_factor: 0.4
+          {# 4 instances in daytime, 2 overnight #}
+          scale_factor: 0.5
```

Comment:
Is this going to be 4 instances in the daytime and 2 overnight? Based on looking at https://github.com/alphagov/notifications-paas-autoscaler/blob/master/app/schedule_scaler.py it looks like its values will either be the minimum (1) or scale factor * max (2)? Is that correct?

Reply:
yeah just realised your interpretation is right, i'd misread the ScheduleScaler.

ScheduleScaler:

* outside of defined hours, set to `min_instances`
* inside of defined hours, set to `max_instances * scale_factor`

Then if you have any other scalers, paas-autoscaler sets to the highest proposed value from all scalers. So if, eg, a, cpu or request or queue scaler or whatever exists, that might set higher, but it'll never all below that max * scale during working hours.

New proposal (to achieve the same goal of 2 overnight, 4 during the day)

* min_instances to 2
* max_instances to 4
* scale_factor to 1.

Reply:
Yep, that looks good.

I've definitely had trouble figuring out how the ScheduleScaler works in the past and feel like it's a candidate to be refactored slightly, maybe at the very least in terms of the name `scale_factor` which I think should actually change to be `scale_factor * min_instances` so you are instead saying run at my min overnight and then during the workday, bump up my minimum to handle higher daytime traffic. And if you want to, then you can scale up even higher based on load/cpu etc.

One for another day though...

Reply:
tbh i want to refactor the whole config (to move the per-space min+max vals from makefile into the yml). and make it live in redis somewhere where we can update it via an admin interface to quickly iterate on things.

oh to have free time